### PR TITLE
Use the 'summariser' matchers in component specs

### PR DESCRIPTION
**Pull request by frankieroberto**
About a year ago I worked with a developer to add a [custom rspec matcher for summary list components](https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/spec/support/summarise_matcher.rb). This aims to avoid having to make a bunch of assertions based on long CSS class names, which makes the specs easier to read.

It also also be used to check any 'actions' (usually 'Change' links) and can use the full text including the visually-hidden text, which helps make sure that the links remain unique and accessible.

Unfortunately we didn't do a great job of telling people about this custom matcher, so it hasn't been widely used! 🤦‍♂️ 

This PR switches a couple of existing specs (picked at random) to use it, as a quick demonstrator.

It'd be great to get feedback on:

* is this custom matcher useful?
* could it be improved in any way?
* are there any other custom matchers we should consider?

### CODE REVIEW

File: spec/components/candidate_interface/english_foreign_language/ielts_review_component_spec.rb [hunk 1]
```diff
@@ -35,8 +65,13 @@
     )
     result = render_inline(described_class.new(ielts_qualification, return_to_application_review: true))
```
Comment (carlosmartinez) [inline spec/components/candidate_interface/english_foreign_language/ielts_review_component_spec.rb:66]: this line can be deleted (and the linter won't be happy until it is)

Comment (frankieroberto) [reply spec/components/candidate_interface/english_foreign_language/ielts_review_component_spec.rb]: Thanks. Missed that one! 🤦

**Review by tomas-stefano — APPROVED** [review]

**Review by carlosmartinez — APPROVED** [review]

**Review by elliotcm — APPROVED** [review]
Thanks @frankieroberto

**Comment by elliotcm** [comment]
> is this custom matcher useful?

Looks to be, yeah.

> could it be improved in any way?

Will have to see once I've used it a bit.

> are there any other custom matchers we should consider?

Probably!  I usually write them for things that are annoying me.

**Pull request merged**